.github/workflows/deploy.yml

```
@@ -0,0 +1,139 @@
+  IMAGE_TAG: ${{ github.sha }}
+
+jobs:
+  run-on-deploy-branch:
+    if: github.head_ref != '' && startsWith(github.head_ref, 'deploy/')
+    runs-on: ubuntu-latest
+    env:
+      SPRING_PROFILES_ACTIVE: test
+
+    steps:
+      - name: Echo branch name
+        run: echo "Running because head_ref is ${{ github.head_ref }}"
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)